### DISCUSSION TITLE
OracleContainer with fixed ports

### DISCUSSION THREAD

**Discussion by buniql** [testcontainers/testcontainers-java#6433] [Q&A] [answered]
Hey,
Is there any way to create an OracleContainer using a fixed port? I've seen some use the deprecated FixedHostPortGenericContainer, but I'd like to use the methods of the OracleContainer.  I know this is discouraged, but I don't think there is another way for me to implement testcontainers without changing up many tests and config files.

The reasoning is that I use system properties to store configurations for several services related to the oracle db (like a flyway migration etc.). I want to set everything needed for those services to make a connection (jdbcURL + user + password) in the system properties. But because this has to happen in runtime after the oracle container has started, I have some errors because the system properties are not set in time. That's why I would like to use a fixed port.

Any help is highly appreciated!

Thanks :)

**Comment by kiview** [comment, marked as answer]
Hey @buniql, using fixed ports is not recommended and not supported. 

In general, there should be no technical reason why you have to do a setup that relies on hardcoded values, rather than doing it at runtime. We have plenty of users that do such setups involving e.g. Spring-Boot, Flyway and RDMS using different dynamic mechanisms to configure it, such as Spring's [`DynamicPropertySource`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/context/DynamicPropertySource.html) or dynamic setting of JVM properties.

**Reply by buniql** [reply]
Thanks for the fast reply. I've done some more research on that topic and concluded it was related to [https://github.com/arquillian/arquillian-core/issues/388](https://github.com/arquillian/arquillian-core/issues/388), which you already helped to resolve. Thanks again!

**Reply by kiview** [reply]
Great to hear, thanks for sharing 🙌
